src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
+ *     description: Fetches a paginated and sorted list of posts created by the user identified by the `userID` parameter.
+ *     parameters:
+ *       - in: path
+ *         name: userID
+ *         required: true
+ *         schema:
+ *           type: string
+ *         description: The unique identifier of the user whose posts are to be retrieved.
+ *       - in: query
+ *         name: perPage
+ *         schema:
+ *           type: integer
+ *           default: 10
+ *           maximum: 20
+ *         description: Number of posts to return per page (up to a maximum of 20).
+ *       - in: query
+ *         name: page
+ *         schema:
+ *           type: integer
+ *           default: 1
+ *         description: Page number to retrieve.
+ *       - in: query
+ *         name: sort
+ *         schema:
+ *           type: string
+ *           default: createdAt
+ *         description: Field by which to sort the results.
+ *       - in: query
+ *         name: reverse
+ *         schema:
+ *           type: string
+ *           enum: [yes, no]
+ *           default: no
+ *         description: Whether to sort results in descending order.
+ *     responses:
+ *       200:
+ *         description: A paginated and sorted list of posts created by the specified user.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: array
+ *               items:
+ *                 type: object
+ *                 properties:
+ *                   _id:
+ *                     type: string
+ *                     description: The unique identifier of the post.
+ *                   title:
+ *                     type: string
+ *                     description: The title of the post.
+ *                   content:
+ *                     type: string
+ *                     description: The content of the post.
+ *                   createdAt:
+ *                     type: string
+ *                     format: date-time
+ *                     description: The creation timestamp of the post.
+ *                   creator_id:
+ *                     type: object
+ *                     properties:
+ *                       _id:
+ *                         type: string
+ *                         description: The unique identifier of the creator.
+ *                       username:
+ *                         type: string
+ *                         description: The username of the post creator.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 err:
+ *                   type: string
+ *                   description: Error message.
+ */
 router.get('/userPosts/:userID', getUserPosts);
+
+/**
+ * @swagger
+ * /posts/count/categories:
+ *   get:
+ *     tags:
+ *       - Posts operations
+ *     summary: Count posts by category
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"